stripe.go

```diff
@@ -485,3 +517,44 @@ func initUserAgent() {
 	}
 	encodedStripeUserAgent = string(marshaled)
 }
+
+// Checks if an error is a problem that we should retry on. This includes both
+// socket errors that may represent an intermittent problem and some special
+// HTTP statuses.
+func (s *BackendConfiguration) shouldRetry(err error, resp *http.Response, numRetries int) bool {
+	if numRetries >= s.MaxNetworkRetries {
+		return false
+	}
+
+	if err != nil {
+		return true
+	}
+
+	if resp.StatusCode == http.StatusConflict {
+		return true
+	}
+	return false
+}
+
+//sleepTime calculates sleeping/delay time in milliseconds between failure and a new one request.
+func (s *BackendConfiguration) sleepTime(numRetries int) (delay int) {
+	// Apply exponential backoff with minNetworkRetryDelay on the
+	// number of num_retries so far as inputs.
+	delay = minNetworkRetryDelay + minNetworkRetryDelay*numRetries*numRetries
```

Comment:
It looks like you were using stripe-ruby's implementation as a template to build this, but that you deviated in a few ways from it  — would you mind just explaining the rationale for why? It's not too big of a deal, but of course maintenance will be slightly easier if all of these are using the same algorithm.

Also, it might be somewhat nice just to make a note in here that we expect the delay in here to be measured in milliseconds and how given that, it's probably okay to calculate randomness with an integer.

Reply:
I think this logic is more clear for go programmer. Also I ruby version delay time could be bigger then maxDelay (because of jitter). I don't think that it's very good. Just say if it's not clear and I should do the identical logic as in ruby.

Reply:
Yeah agreed that it shouldn't go past max delay. Okay, it's fine to leave as is.